nox/_parametrize.py

```diff
@@ -13,9 +13,59 @@
 # limitations under the License.
 
 import functools
+import itertools
 
 
-def parametrize_decorator(arg_names, arg_values_list):
+class Param:
+    """A class that encapsulates a single set of parameters to a parametrized
+    session."""
+
+    def __init__(self, *args, arg_names=None, id=None):
+        self.args = tuple(args)
+        self.id = id
+
+        if arg_names is None:
+            arg_names = ()
+
+        self.arg_names = tuple(arg_names)
+
+    @property
+    def call_spec(self):
+        return dict(zip(self.arg_names, self.args))
+
+    def __str__(self):
+        if self.id:
```

Comment:
Should this be a `None`-check or is `id == ''` also bad? (Won't be particularly readable if someone does set an ID to `''`)

Reply:
yeah "emptiness" test is sufficient. If somehow they pass an empty string I don't want to explode.

Reply:
:+1: